geonode/security/middleware.py

```diff
@@ -32,6 +32,12 @@
 
 from guardian.shortcuts import get_anonymous_user
 
+
+# make sure login_url can be mapped to redirection URL and will match request.path
+login_url = settings.LOGIN_URL.replace(settings.SITEURL.rstrip('/'), '')
```

Comment:
So, we are making the assumption that `LOGIN_URL` must be always defined into the `settings`, but this is a regression if compared to the check removed below:

`getattr(settings, "LOGIN_URL", reverse("account_login"))`

Reply:
@afabiani LOGIN_URL is always provided by Django. The only case where this could break is if a user forcly unset the settings, which is quite unlikely...

Reply:
Uhm, so, should we remove the following from `geonode.settings`?

```python
# Login and logout urls override
LOGIN_URL = os.getenv('LOGIN_URL', '{}account/login/'.format(SITEURL))
LOGOUT_URL = os.getenv('LOGOUT_URL', '{}account/logout/'.format(SITEURL))
```

Reply:
Ok, ignore my last comment... it looks like those are intentional overrides.
Probably they are there for some reason.